### PR TITLE
For conversation

### PR DESCRIPTION
**Pull request by philoserf**
- gofmt
- cleanup markdown
- improve names for less stutter
- use newer context over x/net/context
- clear some lint

### CODE REVIEW

**Comment by philoserf** [comment]
https://youtu.be/cAWlv2SeQus?t=16m

**Comment by alejandroEsc** [comment]
Ha didnt notice this, jump ahead of the things i was working on. Keep in mind this is just poc plus something initially written fast for a demo. If you are still interested in applying a few of your changes, rebase if you dont mind.

File: docs/api.md [hunk 1]
```diff
@@ -1,92 +1,66 @@
 # Protocol Documentation
```
Comment (alejandroEsc) [inline docs/api.md:1]: this is auto-generated.

File: readme.md [hunk 2]
```diff
@@ -1,50 +1,59 @@
-# Setting Up Your Environment
+# gRPC Example
+
+## Setting Up Your Environment
+
 Keep in mind that gRPC requires golang v1.6 or greater, to install run
 
-```
+``` shell
```
Comment (alejandroEsc) [inline readme.md:7]: what does shell buy you here?

File: server/server.go [hunk 3]
```diff
@@ -1,41 +1,47 @@
...
+		"Failed to knock message.")
 )
 
 type doorServer struct {
 	knockFailureMsg string
 }
 
 // Implemented the server interface
-func (s *doorServer) GetHello(c context.Context, knock *a.Knock) (*a.Reply, error) {
+func (s *doorServer) Knocked(c context.Context, k *a.Knock) (*a.Reply, error) {
```
Comment (alejandroEsc) [inline server/server.go:34]: both are not terribly good names. Again POC.

**Review by alejandroEsc — CHANGES_REQUESTED** [review]
please rebase.

File: readme.md (hunk 2 above)
Comment (philoserf) [reply readme.md:7]: It instructs markdown rendering engines about the langauge in the fenced code block. 

https://help.github.com/articles/creating-and-highlighting-code-blocks/

File: docs/api.md (hunk 1 above)
Comment (philoserf) [reply docs/api.md:1]: ACK

**Comment by philoserf** [comment]
no conversation so no reason to keep this

**Pull request closed without merging**